cluster_management/src/main/java/com/pinterest/rocksplicator/LeaderFollowerStateModelFactory.java

```diff
@@ -331,6 +332,37 @@ public void onBecomeFollowerFromLeader(Message message, NotificationContext cont
       Utils.logTransitionCompletionMessage(message);
     }
 
+  public Map<String, String> getLiveHostAndRole(NotificationContext context, String dbName) {
+    HelixAdmin admin = context.getManager().getClusterManagmentTool();
+    ExternalView view = admin.getResourceExternalView(cluster, resourceName);
+    Map<String, String> stateMap = view.getStateMap(partitionName);
+
+    // find live replicas
+    Map<String, String> liveHostAndRole = new HashMap<>();
+    for (Map.Entry<String, String> instanceNameAndRole : stateMap.entrySet()) {
+      String role = instanceNameAndRole.getValue();
+      if (!role.equalsIgnoreCase("LEADER") &&
+          !role.equalsIgnoreCase("FOLLOWER") &&
+          !role.equalsIgnoreCase("OFFLINE")) {
+        continue;
+      }
+
+      String hostPort = instanceNameAndRole.getKey();
+      String host = hostPort.split("_")[0];
+      int port = Integer.parseInt(hostPort.split("_")[1]);
+
+      if (this.host.equals(host)) {
+        // myself
+        continue;
+      }
+
+      if (Utils.getLatestSequenceNumber(dbName, host, port) != -1) {
+        liveHostAndRole.put(hostPort, role);
+      }
```

Comment:
do u recall why we check this seq num here?

Reply:
Not sure of the exact scenario, but it seems to be filter out hosts which are not accessible or dead. But those should be reflected in the external view too.